django_ariadne_jwt/backends.py

```diff
@@ -1,43 +1,164 @@
 """GraphQL auth backends module"""
+import datetime
 from django.contrib.auth import get_user_model
-from .exceptions import JSONWebTokenError
-from .utils import decode_jwt
+from django.conf import settings
+from django.utils import timezone
+from django.utils.module_loading import import_string
+from django.utils.translation import ugettext_lazy as _
+import jwt
+from jwt.exceptions import DecodeError, ExpiredSignatureError
+
+from .exceptions import (
+    AuthenticatedUserRequiredError,
+    ExpiredTokenError,
+    InvalidTokenError,
+    JSONWebTokenError,
+    MaximumTokenLifeReachedError,
+)
+
+
+def load_backend():
+    return import_string(
+        getattr(
+            settings,
+            "JWT_BACKEND",
+            "django_ariadne_jwt.backends.JSONWebTokenBackend",
+        )
+    )()
```

Comment:
How does this code interact with the value from `settings.AUTHENTICATION_BACKENDS`? Wouldn't it allow a configuration issue where two different backends could be specified?

Reply:
This doesn't interact with the `settings.AUTHENTICATION_BACKENDS` code in any way other than having the one object that's pulling double duty and that I'm using the same pattern for loading backends. I could potentially see some issues if someone's got a backend that either:

- Does a lot of heavy lifting in the `__init__`
- Requires that only one instance of the backend class was instantiated

Both of those feel like anti-patterns, so I'm not sure it'd worth coding around that.

---

All that said, I do wonder if it's confusing to folks to see the same backend configuration being used in multiple different areas. I could definitely see making two different parts, but things like loading users and such are shared in both concerns. That was my original thought for combining them.

Reply:
Assuming there is no case for using two different `JSONWebTokenBackend` classes in `settings.AUTHENTICATION_BACKENDS`, maybe we could make `load_backend` detect the JSONWebTokenBackend class or subclass referenced by the settings in order to avoid the duplicity.